pkgs/development/ocaml-modules/cstruct/default.nix

```diff
@@ -1,20 +1,29 @@
-{stdenv, writeText, fetchurl, ocaml, ocplib-endian, sexplib_p4, findlib,
- async_p4 ? null, lwt ? null, camlp4}:
+{stdenv, buildOcaml, fetchFromGitHub, writeText, ocaml, ocplib-endian, sexplib_p4, findlib,
+ async_p4 ? null, lwt ? null, camlp4, ppx_tools ? null}:
```

Comment:
It’ a lot nicer to explicitly create switches for those, then sane defaults can be enforced directly in the package function:

```nix
{ asyncSupport ? false, async_p4 ? null
, … }:

with stdenv.lib;
assert asyncSupport -> async_p4 != null;
…
  configureFlags = optional asyncSupport async_p4 ++ …
```